packages/vm/lib/state/stateManager.ts

```diff
@@ -699,12 +705,17 @@ export default class DefaultStateManager implements StateManager {
     const accessList: AccessList = []
     folded.forEach((slots, addressStr) => {
       const address = Address.fromString(`0x${addressStr}`)
-      if (!address.isPrecompileOrSystemAddress()) {
+      const check1 = address.isPrecompileOrSystemAddress()
```

Comment:
Ah, this is not correct, due to my comment earlier. Only precompiles which have precompiled code ("active precompiles") are already in the access list by default. It will also remove 0x00..00ffff for instance. 

The actual precompiles which you want to remove can be found here: https://github.com/ethereumjs/ethereumjs-monorepo/blob/2c711c76d3371cad32617390e14bc7c4936ffcb8/packages/vm/lib/runTx.ts#L194

Reply:
I don't think this is a practical problem, even [EIP-1352](https://eips.ethereum.org/EIPS/eip-1352) itself builds on the extremely low probability that this address space is currently in use:

> Due to the extremely low probability (and lack of adequate testing possibilities) no explicit checks should be added to ensure that external transaction signing or the invoking of the CREATE instruction can result in a precompile address.

Reply:
(and precompiles which don't exist yet won't be on the initial access list anyhow, or am I missing something?)

Reply:
The edge case here is if someone calls `BALANCE` on 0x00..00ffff. Due to the current implementation, the item will be removed from the access list. But, as you mentioned, since this address is not on the initial access list, we will actually pay the cold read account gas cost for it. If we instead had kept it on the access list, we would have this ~10% gas discount.

Reply:
This is not a super big problem since this will probably never happen in practice, but it is not "correct".

Reply:
Hmm, I am not convinced that it's worth it but have updated 80a9e19, want to get this release out. 😄

Reply:
(so would be great if you can also judge on the solution on @alcuadrado 's request and approve if you are generally satisfied, now with `berlin` so close to `mainnet` I think every day counts)